tests/integration-tests/remote_command_executor.py

```diff
@@ -27,8 +27,10 @@ def __init__(self, result):
 class RemoteCommandExecutor:
     """Execute remote commands on the cluster master node."""
 
-    def __init__(self, cluster):
-        username = get_username_for_os(cluster.os)
+    def __init__(self, cluster, actual_os=None):
```

Comment:
Would it be make more sense to change `actual_os` to something like `execute_as_user`? The caller of this function will take care of figuring out the correct user?
In this context what we really care about is executing command as the right user, but not executing different commands that are dependent on OS, so I would not pass OS in for this reason.